src/cephadm/cephadm

```diff
@@ -2639,10 +2642,11 @@ def command_bootstrap():
                 f.write(ssh_pub)
             logger.info('Wrote public SSH key to to %s' % args.output_pub_ssh_key)
 
-            logger.info('Adding key to root@localhost\'s authorized_keys...')
-            if not os.path.exists('/root/.ssh'):
-                os.mkdir('/root/.ssh', 0o700)
-            auth_keys_file = '/root/.ssh/authorized_keys'
+            logger.info('Adding key to %s@localhost\'s authorized_keys...' % args.ssh_user)
+            ssh_path = 'home/%s' % args.ssh_user if args.ssh_user != 'root' else args.ssh_user
```

Comment:
```suggestion
            user_home_dir = pathify('~%s' % (args.ssh_user))
            if not os.path.exists(user_home_dir):
                raise Error('HOME dir for user %s does not exist: %s'
                    % (args.ssh_user, user_home_dir))
```

The `pathify` function could be used for this, just need to ensure the user/dir actually exists...

per python doc:
```
    expanduser(path)
        Expand ~ and ~user constructions.  If user or $HOME is unknown,
        do nothing.
```